### PR TITLE
SEP for deprecating esxi execution modules

### PR DESCRIPTION
**Pull request by dmurphy18**

### CODE REVIEW

**Review by waynew — APPROVED** [review]

File: deprecate_esxi_modules.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+[design]: #detailed-design
+
+Design is already active in VMware extensions.
+Updates to Salt's documentation for the execution modules for esxi, esxcluster, exsdatacenter and esxvm directing
+users to VMware extensions documentation will need to be made.
+
+## Alternatives
+[alternatives]: #alternatives
+
+Alternative would be to have leave things as they are leading to confusion as to which execution module to use.
```
Comment (chruck) [inline deprecate_esxi_modules.md:34]:
```suggestion
Alternative would be to leave things as they are leading to confusion as to which execution module to use.
```

File: deprecate_esxi_modules.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Unresolved questions
+[unresolved]: #unresolved-questions
+
+What parts of the execution modules for esxi, esxcluster, exsdatacenter and esxvm that are not provided by VMware extensions.
+Requires a search through the more recent VMware releases for features supported, but given that the execution modules for
+esxi, esxcluster, exsdatacenter and esxvm have not had much development since 2016 and only offer basic support, the search
+should only be an excerise in ensuring no surprises.
```
Comment (chruck) [inline deprecate_esxi_modules.md:42]:
```suggestion
should only be an exercise in ensuring no surprises.
```

File: deprecate_esxi_modules.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Why should we *not* do this?
+
+- Cost of migratign existing Salt installations:
+    There will be a cost to migrating existing installations.
+        However most of those users are probably VMware customers and there will be a period of time allowing them
+        to migrate, and the enhanced and supported features of the VMware extensions should give encouragment to migrate.
+
+ - Documentation for Salt.
+    There will be work in updating the exisitng execution modules for esxi, esxcluster, exsdatacenter and esxvm documentation
```
Comment (chruck) [inline deprecate_esxi_modules.md:55]:
```suggestion
    There will be work in updating the existing execution modules for esxi, esxcluster, exsdatacenter and esxvm documentation
```

File: deprecate_esxi_modules.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    but it should be trival to add a line of text and a link redirecting users to the VMware extension documentation.
```
Comment (chruck) [inline deprecate_esxi_modules.md:56]:
```suggestion
    but it should be trivial to add a line of text and a link redirecting users to the VMware extension documentation.
```

**Comment by dmurphy18** [comment]
Closing this since it is basically already covered by https://github.com/saltstack/salt-enhancement-proposals/pull/67, migrating modules to Salt extensions.

**Pull request closed without merging**